### PR TITLE
Remove pointers from default golang types

### PR DESCRIPTION
**Pull request by kostyantyn**
This is a benchmark test I used to measure the performance:

```
var data = []byte(`{"id":"7957b786c3b6426a7c0cbbd10df40fb3","imp":[{"id":"85ef8c62-23ad-4498-73e1-c0aee372999d","native":{"request":"{\"ver\":\"1.0\",\"plcmtcnt\":10,\"assets\":[{\"id\":1,\"required\":1,\"data\":{\"type\":12}},{\"id\":2,\"required\":1,\"title\":{\"len\":50}},{\"id\":3,\"required\":1,\"img\":{\"type\":1,\"w\":80,\"h\":80}},{\"id\":4,\"required\":1,\"img\":{\"type\":3,\"w\":1200,\"h\":627}},{\"id\":5,\"required\":0,\"data\":{\"type\":3}},{\"id\":6,\"required\":1,\"data\":{\"type\":2,\"len\":100}}]}","ver":"1.0"},"bidfloor":3,"bidfloorcur":"USD"},{"id":"adb65948-4480-48c2-4277-4e07ade2930b","native":{"request":"{\"ver\":\"1.0\",\"plcmtcnt\":10,\"assets\":[{\"id\":1,\"required\":1,\"data\":{\"type\":12}},{\"id\":2,\"required\":1,\"title\":{\"len\":50}},{\"id\":3,\"required\":1,\"img\":{\"type\":1,\"w\":80,\"h\":80}},{\"id\":4,\"required\":1,\"img\":{\"type\":3,\"w\":1200,\"h\":627}},{\"id\":5,\"required\":0,\"data\":{\"type\":3}},{\"id\":6,\"required\":1,\"data\":{\"type\":2,\"len\":100}}]}","ver":"1.0"},"bidfloor":3,"bidfloorcur":"USD"},{"id":"17ebb1da-6f67-47e9-54b7-1fa7b599d1e1","native":{"request":"{\"ver\":\"1.0\",\"plcmtcnt\":10,\"assets\":[{\"id\":1,\"required\":1,\"data\":{\"type\":12}},{\"id\":2,\"required\":1,\"title\":{\"len\":50}},{\"id\":3,\"required\":1,\"img\":{\"type\":1,\"w\":80,\"h\":80}},{\"id\":4,\"required\":1,\"img\":{\"type\":3,\"w\":1200,\"h\":627}},{\"id\":5,\"required\":0,\"data\":{\"type\":3}},{\"id\":6,\"required\":1,\"data\":{\"type\":2,\"len\":100}}]}","ver":"1.0"},"bidfloor":3,"bidfloorcur":"USD"},{"id":"023d7d3d-2db7-4505-7975-623dffb9079c","native":{"request":"{\"ver\":\"1.0\",\"plcmtcnt\":10,\"assets\":[{\"id\":1,\"required\":1,\"data\":{\"type\":12}},{\"id\":2,\"required\":1,\"title\":{\"len\":50}},{\"id\":3,\"required\":1,\"img\":{\"type\":1,\"w\":80,\"h\":80}},{\"id\":4,\"required\":1,\"img\":{\"type\":3,\"w\":1200,\"h\":627}},{\"id\":5,\"required\":0,\"data\":{\"type\":3}},{\"id\":6,\"required\":1,\"data\":{\"type\":2,\"len\":100}}]}","ver":"1.0"},"bidfloor":3,"bidfloorcur":"USD"},{"id":"96cf5aab-c5dd-4bb3-5a83-1db7999e5c4e","native":{"request":"{\"ver\":\"1.0\",\"plcmtcnt\":10,\"assets\":[{\"id\":1,\"required\":1,\"data\":{\"type\":12}},{\"id\":2,\"required\":1,\"title\":{\"len\":50}},{\"id\":3,\"required\":1,\"img\":{\"type\":1,\"w\":80,\"h\":80}},{\"id\":4,\"required\":1,\"img\":{\"type\":3,\"w\":1200,\"h\":627}},{\"id\":5,\"required\":0,\"data\":{\"type\":3}},{\"id\":6,\"required\":1,\"data\":{\"type\":2,\"len\":100}}]}","ver":"1.0"},"bidfloor":3,"bidfloorcur":"USD"},{"id":"d856c711-8f69-4ee5-5a42-e1d4ea25ea24","native":{"request":"{\"ver\":\"1.0\",\"plcmtcnt\":10,\"assets\":[{\"id\":1,\"required\":1,\"data\":{\"type\":12}},{\"id\":2,\"required\":1,\"title\":{\"len\":50}},{\"id\":3,\"required\":1,\"img\":{\"type\":1,\"w\":80,\"h\":80}},{\"id\":4,\"required\":1,\"img\":{\"type\":3,\"w\":1200,\"h\":627}},{\"id\":5,\"required\":0,\"data\":{\"type\":3}},{\"id\":6,\"required\":1,\"data\":{\"type\":2,\"len\":100}}]}","ver":"1.0"},"bidfloor":3,"bidfloorcur":"USD"},{"id":"4198c960-4ddb-4b3e-6e8d-9c42d7064cba","native":{"request":"{\"ver\":\"1.0\",\"plcmtcnt\":10,\"assets\":[{\"id\":1,\"required\":1,\"data\":{\"type\":12}},{\"id\":2,\"required\":1,\"title\":{\"len\":50}},{\"id\":3,\"required\":1,\"img\":{\"type\":1,\"w\":80,\"h\":80}},{\"id\":4,\"required\":1,\"img\":{\"type\":3,\"w\":1200,\"h\":627}},{\"id\":5,\"required\":0,\"data\":{\"type\":3}},{\"id\":6,\"required\":1,\"data\":{\"type\":2,\"len\":100}}]}","ver":"1.0"},"bidfloor":3,"bidfloorcur":"USD"},{"id":"93dc786b-cfc1-4fff-5617-39687aed77ec","native":{"request":"{\"ver\":\"1.0\",\"plcmtcnt\":10,\"assets\":[{\"id\":1,\"required\":1,\"data\":{\"type\":12}},{\"id\":2,\"required\":1,\"title\":{\"len\":50}},{\"id\":3,\"required\":1,\"img\":{\"type\":1,\"w\":80,\"h\":80}},{\"id\":4,\"required\":1,\"img\":{\"type\":3,\"w\":1200,\"h\":627}},{\"id\":5,\"required\":0,\"data\":{\"type\":3}},{\"id\":6,\"required\":1,\"data\":{\"type\":2,\"len\":100}}]}","ver":"1.0"},"bidfloor":3,"bidfloorcur":"USD"},{"id":"afc27f69-179a-4118-4b5f-ee621f260083","native":{"request":"{\"ver\":\"1.0\",\"plcmtcnt\":10,\"assets\":[{\"id\":1,\"required\":1,\"data\":{\"type\":12}},{\"id\":2,\"required\":1,\"title\":{\"len\":50}},{\"id\":3,\"required\":1,\"img\":{\"type\":1,\"w\":80,\"h\":80}},{\"id\":4,\"required\":1,\"img\":{\"type\":3,\"w\":1200,\"h\":627}},{\"id\":5,\"required\":0,\"data\":{\"type\":3}},{\"id\":6,\"required\":1,\"data\":{\"type\":2,\"len\":100}}]}","ver":"1.0"},"bidfloor":3,"bidfloorcur":"USD"},{"id":"943d36fe-d7e2-499f-4414-4fa22d38453c","native":{"request":"{\"ver\":\"1.0\",\"plcmtcnt\":10,\"assets\":[{\"id\":1,\"required\":1,\"data\":{\"type\":12}},{\"id\":2,\"required\":1,\"title\":{\"len\":50}},{\"id\":3,\"required\":1,\"img\":{\"type\":1,\"w\":80,\"h\":80}},{\"id\":4,\"required\":1,\"img\":{\"type\":3,\"w\":1200,\"h\":627}},{\"id\":5,\"required\":0,\"data\":{\"type\":3}},{\"id\":6,\"required\":1,\"data\":{\"type\":2,\"len\":100}}]}","ver":"1.0"},"bidfloor":3,"bidfloorcur":"USD"}],"app":{"id":"1008836","name":"JustMusicPlayer","cat":["IAB1"],"bundle":"justmoby.justmusic.player","publisher":{"id":"1005077"},"keywords":"player,music,song,songs,media,ringtone,sound,radio","storeurl":"https://play.google.com/store/apps/details?id=justmoby.justmusic.player"},"device":{"dnt":2,"ua":"Dalvik/2.1.0(Linux;U;Android5.1;XT1254Build/SU4TL-44)","ip":"70.192.40.36","geo":{"lat":42.3955,"lon":-71.1816,"country":"USA","region":"MA","city":"Belmont","zip":"02478","type":2},"carrier":"VerizonWireless","language":"en","model":"XT1254","os":"Android","osv":"5.1","connectiontype":3,"devicetype":1,"ifa":"798ec462-b57b-48c6-a53b-f10efde107c0"},"user":{"id":"63a4b23e-7ebd-5bdd-a501-4e939f7c4048"},"at":2,"tmax":100,"wseat":[""],"allimps":0,"cur":["USD"]}`)
func BenchmarkRequestPerformance(t *testing.B) {
	for i := 0; i < t.N; i++ {
		req, err := ParseRequestBytes(data)
		assert.Nil(t, err)
		_, err = json.Marshal(req)
		assert.Nil(t, err)
	}
}
```

Current implementation:
`BenchmarkRequestPerformance-4	   10000	    202965 ns/op`
Without pointers:
`BenchmarkRequestPerformance-4	   10000	    191501 ns/op`

Also, we reduce the heap size, so it's another extra improvement.

### CODE REVIEW

**Comment by bruwozniak** [comment]
looks :ok:

File: banner.go [hunk 1]
```diff
@@ -7,48 +7,24 @@ package openrtb
...
-	Wmin     *int       `json:"wmin,omitempty"`     // Width minimum
-	Hmin     *int       `json:"hmin,omitempty"`     // Height minimum
-	Id       *string    `json:"id,omitempty"`       // A unique identifier
-	Pos      *int       `json:"pos,omitempty"`      // Ad Position
+	W        int        `json:"w,omitempty"`        // Width
+	H        int        `json:"h,omitempty"`        // Height
+	Wmax     int        `json:"wmax,omitempty"`     // Width maximum
+	Hmax     int        `json:"hmax,omitempty"`     // Height maximum
+	Wmin     int        `json:"wmin,omitempty"`     // Width minimum
+	Hmin     int        `json:"hmin,omitempty"`     // Height minimum
```
Comment (miry) [inline banner.go]: btw this is deprecated and not used in the 2.4

**Comment by miry** [comment]
:+1:

File: banner.go (hunk 1 above)
Comment (kostyantyn) [inline banner.go]: let's keep it for now. we would need to update it later when we start working on 2.4.

**Pull request merged**